jsonschema-core.xml

```diff
@@ -288,6 +278,36 @@
                         zeros) are insignificant.
                     </t>
                 </section>
+
+                <section title="Non-JSON Instances">
+                    <t>
+                        It is possible to use JSON Schema with a superset of the JSON Schema data model,
+                        where an instance may be outside any of the six JSON data types.
+                    </t>
+                    <t>
+                        In this case, annotations still apply; but validation keywords will not be useful,
+                        as they will always pass or always fail.
+                    </t>
+                    <t>
+                        A custom vocabulary may define support for a superset of the core data model.
+                        The schema itself may only be expressible in this superset;
+                        for example, to make use of the "const" keyword.
+                    </t>
+                    <t>
+                        New data model types may be defined as either supersets of an existing type
+                        (similar to how "number" is a superset of "integer"),
+                        or a type disjoint from the core data model altogether (like "binary").
+                        Existing keywords may be updated to define behavior for the new range of values,
+                        but vocabularies that intend to be compatible with JSON Schema Validation should not
```

Comment:
/should not/MUST NOT/ ?

Reply:
I'll have to review the spec again, but I'd guess vocabularies are allowed to not use the validation vocabulary.

I wonder if it would save implementation effort to not allow this?

Reply:
Yes this is true. My feeling is that the basic use of type should be pretty universal, and SHOULD NOT makes it clear you better have a good reason to change the behavior of existing types.

Reply:
I thought @handrews had shown elsewhere in this thread that we would not be permitting updating existing keywords, so MUST NOT is correct.

Reply:
Correct. I read this comment, made a comment, before reading the rest of the discussion related to this issue.

Reply:
@awwright it is definitely true that a _dialect_ can omit the validation _vocabulary._ (I'm trying to promoted the "dialect" term, see #902).

We need to be careful here of the distinction between `type` (the keyword in the validation vocabulary) and the more generic notion of JSON Schema data model.  I'm a little confused by:

> New data model types may be defined as either supersets of an existing type (similar to how "number" is a superset of "integer")

as "number" is both a data model type and a value for `type`, while "integer" is ***not*** a data model type, just a value for `type`.  "integer" is a refinement/shortcut `type` assertion for the data model type "number" plus `"multipleOf": 1` (more or less, I don't recall the exact language for "integer" offhand).

A binary value would be outside of the JSON Schema data model, and therefore be an extension (not refinement or shortcut) to that data model.  Because the `type` keyword only has keywords relevant to the JSON data model, you would not be able to use `type` with an extension to the data model and have the assertion pass.

------

An important consideration here is that when we say "this keyword applies to numbers" or "this keyword applies to strings" we are talking about the JSON data model.  We can't say "this keyword applies to integers" because the only way to determine that something is in integer is to use `type: integer`.  The data model is not aware of integers.

My _slight_ preference after reading all this is to basically add one additional thing to the JSON Schema data model: none of the above.  Let's not get into allowing people to make specific extensions to the data model because the implications of that are potentially complex.  But we can explicitly allow people to use JSON Schema with data that doesn't fit the data model, assuming their implementation supports it somehow.  

With this approach, you could add a keyword that works for "values outside the primary JSON Schema data model" or whatever we want to call it.  But you can't add keywords for various different sorts of things that might show up that way.

In practice with OpenAPI, these "instances" are being documented and handed off to something else to interpret based on the JSON Schema `contentMediaType` keyword.  It's a way to say, _from within JSON Schema_, that the thing is not actually JSON and shouldn't be interpreted as such, but also shouldn't be an error.

(BTW, much thanks to @awwright for picking this up and to everyone for comments- this is going to be a much more solid change than my original idea when it's all done).

Reply:
Follow up thoughts, I often a use case for wanting to validate a JSON structure in javascript which sometimes contains a function... You can't convert them to JSON, but, in js, it could be part of the data you want to validate, where you don't care about what the function is, just that it's a function.